src/core/parser_internal/func_worker.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "docgen_worker.hpp"
+#include "ignore_worker.hpp"
+
+namespace docgen {
+namespace core {
+namespace parser_internal {
+
+struct FuncWorker_routines : private routine_details_t
+{
+	private:
+		static constexpr const char *FUNCS_KEY = "functions";
+		static constexpr const char *DEC_KEY = "declaration";
+
+	protected:
+		static constexpr const routine_t on_dec_ = [](worker_t *, const tok_t&, dest_t& writer) {
+			writer.set_key(DEC_KEY);	
+			writer.start_writing();
+		};
+		static constexpr const routine_t on_done_ = [](worker_t *worker, const tok_t& token, dest_t& writer) {
+			writer.stop_writing();
+			writer.clear_key();
+			writer.store(FUNCS_KEY);
+			if (token.name == Symbol::SEMICOLON) {
+				worker->reset();
+			}
+		};
+};
+
+class FuncWorker : public worker_class_t, private FuncWorker_routines
+{
+	public:
+		FuncWorker()
+			: ParseWorker {
```

Comment:
if you're going to use `worker_class_t`, only use that, so instead of `ParseWorker` it should be `worker_class_t`.

Reply:
I think my reasoning behind this was that once we've inherited from `ParseWorker<LongSpecialization>`, there is thankfully no need to detail the specialization parameters within the class anymore (the compiler figures this out), so I might as well take advantage of the fact that I can just say `ParseWorker` in this scenario and be explicit. This seems like a reasonable distinction—the aliases contain the specializations, but it's not needed in this case. Can change if not though.

Reply:
I see. I think of aliases in this case as an interface just like how we do for member functions in a class. It unifies our type interface such that if implementation changes, we only need to change one place. It is implementation detail that this class is inheriting from a ParseWorker<...> which is reflected by worker_class_t

Reply:
Right... of course, you wouldn't only change in one place, as the constructor used here is specifically that of `ParseWorker`, so updating `worker_internal_t` would require changing these. However, there are other places in my code where I use members specific to `ParseWorker` and `JSONWriter` on type alias variables, so it would be most consistent to have these be type aliases as well. I will update this.